### DISCUSSION TITLE
Hubspot api scraping

### DISCUSSION THREAD

**Discussion by mhensen** [alex-on-ai/WebReaper#30] [Q&A]
Hi, I want to scrape the documentation at the url https://developers.hubspot.com/beta-docs/reference/api, the problem I encounter is that the links on the left are not really povided.. but they come up on clicking each individual div .. is there any apoach I can use to get the data form the  centre? I need this data to put into a rag.. 

regards,

Michael

**Comment by alex-on-ai** [comment]
Hi @mhensen — two ways to handle links that only appear after a client-side interaction:

**1. Drive a real browser and interact, then follow the rendered links:**

```csharp
.GetWithBrowser(new[] { url }, actions => actions
    .Click(".the-expander")
    .WaitForNetworkIdle())
.Follow("a.nav-link")
```

Note: of the page actions, `Click`, `Wait`, `ScrollToEnd` and `WaitForNetworkIdle` are executed by the browser loader; `WaitForSelector` / `EvaluateExpression` exist on the builder but are not currently wired, so prefer a `Wait(...)` or `WaitForNetworkIdle()` after the click rather than `WaitForSelector`.

**2. Skip the UI and hit the data source (recommended for docs sites):**

A reference/docs site like that almost always loads its navigation and content from a JSON API. Open DevTools → Network, find the JSON request, point WebReaper at that endpoint and use `.WithJsonContentParser()` (selectors become JSONPath). For a RAG corpus this is far more robust than clicking divs and scraping rendered HTML.
